packages/typo3-guides-cli/src/Migration/SettingsMigrator.php

```diff
@@ -101,28 +113,49 @@ private function createProjectSection(\DOMElement $parentNode): void
                 unset($this->unmigratedSettings['general'][$option->name]);
             }
         }
-
-        $parentNode->append($project);
+        return $project;
     }
 
     /**
      * Add <inventory> Element. This gets filled with the old "intersphinx_mapping" section.
+     * @return list<\DOMElement>
      **/
-    private function createInventorySection(\DOMElement $parentNode): void
+    private function createInventorySection(): array
     {
+        $inventories = [];
         if (!is_array($this->legacySettings['intersphinx_mapping'] ?? false)) {
-            return;
+            return [];
         }
+        $version = null;
 
         foreach ($this->legacySettings['intersphinx_mapping'] as $id => $url) {
             unset($this->unmigratedSettings['intersphinx_mapping'][$id]);
             $this->convertedSettings++;
 
+            if ($defaultInventory = DefaultInventories::tryFrom($id)) {
+                $defaultUrl = $defaultInventory->getUrl();
+                if ($url === $defaultUrl) {
+                    continue;
+                } else {
+                    $allowedVersions = Typo3VersionMapping::getAllVersions();
+                    $urlDifference = $this->getStringDifference($url, $defaultUrl);
+                    if (in_array($urlDifference, $allowedVersions, true)) {
```

Comment:
Perhaps use Typo3VersionMapping::tryFrom(), this returns the case or null. So no need to call getAllversions() and check for existence of a value.

Reply:
The allowed versions are the result of $versionType->getVersion(), you cannot get them by calling tryFrom